From 165a736c232baf9ff4b070dd7266d7033e450fd9 Mon Sep 17 00:00:00 2001
From: Christian Hergert <chergert@redhat.com>
Date: Wed, 20 Apr 2022 16:04:48 -0700
Subject: [PATCH 2/4] plugins/flatpak: port to libsoup-3.0
X-Developer-Signature: v=1; a=openpgp-sha256; l=3068; i=xtexchooser@duck.com;
 h=from:subject; bh=saafDEmxNpe2zhbwbTLyazUs9j3JaEC309iM8o6aEXU=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDBlWTo+nnNJvdrXfrhx0JnjXArkP819zHeA3MIvxWz/z2
 jRxxi23O0pZGMS4GGTFFFmKDBu8WXXS+UWXlcvCzGFlAhnCwMUpABMpN2T4K7Ckvrcx+87r4mKb
 iylGZ2RKi/Z8uHvjXkiokWhiU5ThLoY/3Lt3dxQdfX5h+fpiu/3rOZfY/G7y1lrdyLN4VXZhwVQ
 DdgA=
X-Developer-Key: i=xtexchooser@duck.com; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

(cherry picked from commit 9bd9426ed220ce015c2d9c3e1b4058e2845323c9)
---
 src/plugins/flatpak/gbp-flatpak-sources.c | 26 +++++++++++------------
 src/plugins/flatpak/meson.build           |  2 +-
 2 files changed, 13 insertions(+), 15 deletions(-)

diff --git a/src/plugins/flatpak/gbp-flatpak-sources.c b/src/plugins/flatpak/gbp-flatpak-sources.c
index c7cc1a686e9d..5fd724bb9f79 100644
--- a/src/plugins/flatpak/gbp-flatpak-sources.c
+++ b/src/plugins/flatpak/gbp-flatpak-sources.c
@@ -439,28 +439,24 @@ create_uncompress_directory (GFile   *dest,
 static SoupSession *
 get_soup_session (void)
 {
-  return soup_session_new_with_options (SOUP_SESSION_USER_AGENT, PACKAGE_NAME,
+  return soup_session_new_with_options ("user-agent", PACKAGE_NAME,
                                         NULL);
 }
 
 static GBytes *
-download_uri (SoupURI  *uri,
-              GError  **error)
+download_uri (GUri    *uri,
+              GError **error)
 {
   g_autoptr(SoupSession) session = NULL;
-  g_autoptr(SoupRequest) req = NULL;
+  g_autoptr(SoupMessage) msg = NULL;
   g_autoptr(GInputStream) input = NULL;
   g_autoptr(GOutputStream) out = NULL;
 
   g_assert (uri != NULL);
 
   session = get_soup_session ();
-
-  req = soup_session_request_uri (session, uri, error);
-  if (req == NULL)
-    return NULL;
-
-  input = soup_request_send (req, NULL, error);
+  msg = soup_message_new_from_uri ("GET", uri);
+  input = soup_session_send (session, msg, NULL, error);
   if (input == NULL)
     return NULL;
 
@@ -476,7 +472,7 @@ download_uri (SoupURI  *uri,
 }
 
 static gboolean
-download_archive (SoupURI      *uri,
+download_archive (GUri         *uri,
                   const gchar  *sha,
                   GFile        *archive_file,
                   GError      **error)
@@ -577,7 +573,7 @@ gbp_flatpak_sources_fetch_archive (const gchar  *url,
 {
   g_autoptr(GFile) archive_file = NULL;
   g_autoptr(GFile) source_dir = NULL;
-  g_autoptr(SoupURI) uri = NULL;
+  g_autoptr(GUri) uri = NULL;
   g_autofree char *archive_name = NULL;
   GError *local_error = NULL;
 
@@ -593,8 +589,10 @@ gbp_flatpak_sources_fetch_archive (const gchar  *url,
       g_error_free (local_error);
     }
 
-  uri = soup_uri_new (url);
-  archive_name = g_path_get_basename (soup_uri_get_path (uri));
+  if (!(uri = g_uri_parse (url, G_URI_FLAGS_NONE, error)))
+    return NULL;
+
+  archive_name = g_path_get_basename (g_uri_get_path (uri));
   archive_file = g_file_get_child (source_dir, archive_name);
 
   if (!download_archive (uri, sha, archive_file, error))
diff --git a/src/plugins/flatpak/meson.build b/src/plugins/flatpak/meson.build
index d6455ab0b867..8424e915d4c2 100644
--- a/src/plugins/flatpak/meson.build
+++ b/src/plugins/flatpak/meson.build
@@ -9,7 +9,7 @@ if not libportal_dep.found()
 endif
 
 libostree_dep = dependency('ostree-1')
-libsoup_dep = dependency('libsoup-2.4', version: '>= 2.52.0')
+libsoup_dep = dependency('libsoup-3.0')
 
 subdir('daemon')
 
-- 
2.49.0

